core/services/relay/evm/functions/config_poller.go

```diff
@@ -175,3 +180,19 @@ func (cp *configPoller) LatestBlockHeight(ctx context.Context) (blockHeight uint
 	}
 	return uint64(latest), nil
 }
+
+// called from LogPollerWrapper in a separate goroutine
+func (cp *configPoller) UpdateRoutes(activeCoordinator common.Address, proposedCoordinator common.Address) error {
+	cp.targetContract.Store(&activeCoordinator)
+	// Register filters for both active and proposed
+	err := cp.destChainLogPoller.RegisterFilter(logpoller.Filter{Name: configPollerFilterName(activeCoordinator), EventSigs: []common.Hash{ConfigSet}, Addresses: []common.Address{activeCoordinator}})
+	if err != nil {
+		return err
+	}
+	err = cp.destChainLogPoller.RegisterFilter(logpoller.Filter{Name: configPollerFilterName(proposedCoordinator), EventSigs: []common.Hash{ConfigSet}, Addresses: []common.Address{activeCoordinator}})
```

Comment:
Why do we need to register a filter for the proposed coordinator?  Aren't we always using the OCR config from the active coordinator contract?

Reply:
To start collecting logs early and later be ready when it becomes the active one. Otherwise we wouldn't see the OCR config updates that happened before the proposal takes effect. (this statement is based on an assumption that my limited understanding of LogPoller is correct).